optimade/server/middleware.py

```diff
@@ -8,13 +8,17 @@ class RedirectSlashedURLs(BaseHTTPMiddleware):
     async def dispatch(self, request: Request, call_next):
         """Redirect URL requests ending with a slash to non-slashed URLs
 
-        E.g., `http://example.org/info/` -> `http://example.org/info`
+        E.g., `http://example.org/optimade/v0/info/` -> `http://example.org/optimade/v0/info`
         """
         if request.scope["path"].endswith("/"):
             redirect_scope = dict(request.scope)
 
-            # Make sure we're not dealing with a URL path (after the domain) of `/`
-            if redirect_scope["path"] != "/":
+            # Make sure to only redirect paths that are beyond the versioned depth
+            # e.g. don't redirect `/optimade/` to `/optimade` as this will loop indefinitely
+            if (
+                len([substr for substr in redirect_scope["path"].split("/") if substr])
+                > 1
```

Comment:
This is quite fragile, since the non-versioned base URL may be something weird, like `http://example.org/some/path/to/optimade`

Maybe we could use `CONFIG.base_url` or something?

Reply:
Hmmm, from my experimenting I was under the impression that the `redirect_scope["path"]` your example case would still just contain `/optimade/`. Will investigate.

Reply:
The path should be everything after the domain, i.e. for the example above it should be `/some/path/to/optimade`.

Reply:
I think it's most sensible if we restrict this to only redirecting the entry endpoints, i.e. those where having the slash or not gives you different behaviour...

Reply:
That might be very true - maybe they can be retrieved dynamically by going through the `app`'s `routes`?

Reply:
I just took them from `optimade.routers.ENTRY_COLLECTIONS` in the end.